ui/src/app/settings/selectors/repositories.js

```diff
@@ -1,10 +1,52 @@
 const repositories = {};
 
 /**
- * Returns all repositories
+ * Returns list of all repositories
  * @param {Object} state - The redux state.
  * @returns {Array} A list of all repositories.
  */
-repositories.get = state => state.repositories.items;
+repositories.all = state => state.repositories.items;
+
+/**
+ * Returns count of all repositories
+ * @param {Object} state - The redux state.
+ * @returns {Number} Count of all repositories.
+ */
+repositories.count = state => state.repositories.items.length;
+
+/**
+ * Returns true if repositories are loading.
+ * @param {Object} state - The redux state.
+ * @returns {Boolean} Repositories are loading.
+ */
+repositories.loading = state => state.repositories.loading;
+
+/**
+ * Returns true if repositories have been loaded.
+ * @param {Object} state - The redux state.
+ * @returns {Boolean} Repositories have loaded.
+ */
+repositories.loaded = state => state.repositories.loaded;
+
+/**
+ * Get repositories that match a term.
+ * @param {Object} state - The redux state.
+ * @param {String} term - The term to match against.
+ * @returns {Array} A filtered list of repositories.
+ */
+repositories.search = (state, term) => {
+  return state.repositories.items.filter(
+    repo => repo.name.includes(term) || repo.url.includes(term)
```

Comment:
Because we override the names of the repos, this won't match against the displayed names. I wonder if there's any way to include those values.

Reply:
Hmm this is a tricky one. In RBAC we had a separate field in state for filter term. That way we could compose a selector for search that would include the filter term and a subset/mutated list of entities (e.g `repositories.getForSettings` or something that just renames the defaults). I've created [a card](https://github.com/canonical-web-and-design/maas-ui/issues/120) because I think a robust solution might take a bit of work with actions/reducers/selectors.